lib/ui/recommendation.js

```diff
@@ -154,53 +159,92 @@ Recommendation.prototype = {
     this.events.publish('recommendation-mousemove', evt);
   },
   show: function(data) {
-    if (this.timeout) {
-      this.win.clearTimeout(this.timeout);
-      this.timeout = null;
+    if (this.hideTimeout) {
+      this.win.console.log('show clearing hideTimeout');
+      this.win.clearTimeout(this.hideTimeout);
+      this.hideTimeout = null;
+    }
+
+    if (!this.el) {
+      return;
     }
 
-    // Don't re-render the recommendation if the content hasn't changed since
-    // the previous recommendation.
-    if (this.prev && this.el && this.el.firstChild && data.result.url === this.prev.result.url) {
+    this.win.console.log('about to dupe check inside show().');
+
+    // Reuse the recommendation if it's still in the DOM and the content hasn't
+    // changed since the previous recommendation.
+    if (this.el.firstChild && data && this.row &&
+        this.row.url === data.url) {
       this.el.collapsed = false;
       this.events.publish('recommendation-shown', this);
+      this.win.console.log('show found a dupe in the DOM, bailing.');
       return;
     }
 
-    this.prev = data;
-    const recommendation = this.row = new this.RecommendationRow(data, {
+    this.data = data;
+
+    this.win.console.log('not a dupe. this.row.url is',this.row && this.row.url,', and data.url is ',data.url);
+    this.row = new this.RecommendationRow(this.data, {
```

Comment:
Do we need to `delete this.row` before creating a new row? Will the old row be properly GC'd?